### ISSUE TITLE
React + Redux architecture

### ISSUE THREAD

**Issue by benja-M-1** [theodo/theodo-stack-generator#51]
I was wondering why the folder containing the global store, reducer and sagas is named ```module```? Moreover, there are modules in pages alongside to components, container, tests and so on... it is even more confusing.

@cRicateau @tcheymol can you explain it to me here please?

**Comment by cRicateau** [comment]
@benja-M-1 From what I have read so far on the subject, my recommandation would be to move toward this pattern: https://medium.freecodecamp.org/scaling-your-redux-app-with-ducks-6115955638be 👍 
```
duck/
├── actions.js
├── index.js
├── operations.js
├── reducers.js
├── selectors.js
├── tests.js
├── types.js
├── utils.js
```

**Comment by benja-M-1** [comment]
I would love to have a broder vue of a react + redux + saga application. I propose the following file tree based on the discussions I had today with @lechinoix.
I know that @ghamaide disagree on the split of reducers + actions + selectors, I don't have opinion on that. I simply think that we should have one and only one way of doing things to avoir the burden to ask ourselves if it is time or not to split the concepts on its own file.

```
src/
|-- components/
    |-- Article/
        |-- Article.container.js
        |-- Article.component.js
        |-- Article.styles.js
    |-- ArticleExcerpt/
        |-- ArticleExcerpt.container.js
        |-- ArticleExcerpt.component.js
        |-- ArticleExcerpt.styles.js
|-- entities/
    |-- Articles/ <-- note the plural form
        |-- Articles.api.js
        |-- Articles.type.js
        |-- Articles.normalize.js
        |-- Articles.sagas.js
        |-- Articles.reducer.js
        |-- Articles.actions.js <-- not defined yet
        |-- Articles.selectors.js <-- not defined yet
    |-- Authors/
        |-- api.js
        |-- type.js
        |-- normalize.js
        |-- sagas.js
        |-- reducer.js
|-- pages/
    |-- Homepage/
        |-- ... here you put routes, page reducers, utils... ?
    |-- Articles/
|-- services/
...
```

@cRicateau @tcheymol @phacks @antkahn @Paulmolin @Bernardstanislas

**Comment by ghamaide** [comment]
@benja-M-1 my pain point is the fact that you have to edit 5 files when you do redux stuff :) In this architecture, where do you put the redux stuff that is used for ui or routing in your app ? Like a side-bar toggle ? 
In `entities/Authors`, I think your file names should be prefixed by Authors like in `entities/Articles`
By the way, if we build SPAs (single page applications), shouldn't our `pages` directory rather be called `views` ? 😇

**Comment by benja-M-1** [comment]
We looked at what BAM does on investo and it is very clear and easy to use.

```
components
pages
redux
 |-- Articles
 |-- Modal
 |-- Toaster
services
```

---
pages : 
* are linked to a route
* it declares every components in the page

components :
* are autonomous UI elements
* there are no reducers in components

**Comment by phacks** [comment]
On Smart Advisor we use an “investo-like” structure and I couldn’t be more happy with it.

**Comment by cRicateau** [comment]
@benja-M-1 resolved #78
